internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -711,24 +742,51 @@ func NewCommandChannelService(cr *operatorv1beta1.Cryostat) *corev1.Service {
 }
 
 func NewGrafanaService(cr *operatorv1beta1.Cryostat) *corev1.Service {
+	// Initialize with defaults
+	svcType := corev1.ServiceTypeClusterIP
+	httpPort := int32(3000)
+	labels := map[string]string{}
+	annotations := map[string]string{}
+
+	// Override with any user-specified properties
+	if cr.Spec.ServiceOptions != nil && cr.Spec.ServiceOptions.GrafanaConfig != nil {
```

Comment:
Can this if-statement pattern and the various variables it assigns be extracted out to a function that returns either multiple values or some structure that has fields for each of the values? Then it can be reused here and in `NewCoreService` - DRY

Reply:
I tried refactoring this into something nicer, but didn't succeed. I'll make another attempt.

Reply:
I think I've come up with a good refactoring of this. What do you think?

Reply:
Looks better, thanks